Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Fleet/Registration.hs

```diff
@@ -161,6 +162,10 @@ fleetOwnerLogin req = do
   merchant <-
     QMerchant.findByShortId merchantId
       >>= fromMaybeM (MerchantNotFound merchantId.getShortId)
+  fleetOwnerInfo <-
+    QFOI.findByMerchantId merchant.id
```

Comment:
Why `findByMerchantId`? One merchant can have many fleet owners
Should be something like this

```
          person <- QP.findByMobileNumberAndMerchantAndRoles req.mobileCountryCode mobileNumberHash merchant.id [DP.FLEET_OWNER, DP.OPERATOR] >>= fromMaybeM (PersonNotFound req.mobileNumber) --because currently fleetOwner and operator can use this api
          when (person.role == DP.FLEET_OWNER) $
            fleetOwnerInfo <- findByPrimaryKey person.id >>= fromMaybeM (PersonDoesNotExist personId.getId)
            unless fleetOwnerInfo.enabled (throwError $ InvalidRequest "fleetOwner is not enabled")
```